src/doc/grammar.md

```diff
@@ -512,7 +512,7 @@ unit_expr : "()" ;
 ```antlr
 struct_expr : expr_path '{' ident ':' expr
                       [ ',' ident ':' expr ] *
-                      [ ".." expr ] '}' |
+                      [ ".." expr ] ? '}' |
```

Comment:
IIRC `[ <rules> ]` already means `<rules>` are optional. 

Side note: this rule says `A { a: 0 .. A { ..Default::default() }` (note the missing comma after the `0`) would do base update twice, but the comma between fields and base update cannot be omitted. It also disallows `A { ..Default::default() }` which is valid, because it requires at least one field. Want to fix that?